src/Spotflow.InMemory.Azure.Storage/Blobs/Internals/InMemoryBlobContainer.cs

```diff
@@ -31,27 +31,89 @@ public BlobContainerProperties GetProperties()
 
     public override string? ToString() => $"{Service} / {Name}";
 
-    public IReadOnlyList<BlobItem> GetBlobs(string? prefix, BlobStates? states)
+    public IReadOnlyList<BlobItem> GetBlobs(string? prefix, BlobTraits traits, BlobStates states)
```

Comment:
Please refactor as follows:

- Avoid throwing (client) exception from this layer. Refactor to `TryGetBlobs(..., out GetBlobsError? error)` form (similar as e.g. in `InMemoryBlockBlob.TryStageBlock`.
- Invert the enum validation - check that only `BlobTraits.Metadata | BlobTraits.None` is present. Similarly for `BlobStates`.
- Move the enum validation to the call site and only pass simple and clear flags: `bool includeUncommittedBlobs`, `bool includeMetadata`